descartes_moveit/src/moveit_state_adapter.cpp

```diff
@@ -247,11 +246,23 @@ bool MoveitStateAdapter::isInCollision(const std::vector<double>& joint_pose) co
   bool in_collision = false;
   if (check_collisions_)
   {
-    moveit::core::RobotState state (robot_model_ptr_);
-    state.setToDefaultValues();
-    state.setJointGroupPositions(joint_group_, joint_pose);
-    in_collision = planning_scene_->isStateColliding(state, group_name_);
+    std::unique_ptr<planning_scene_monitor::LockedPlanningSceneRO> ls(new planning_scene_monitor::LockedPlanningSceneRO(planning_scene_monitor_));
+
+    // Check if we successfully got a locked planning scene
+    if (!(*ls))
+    {
+      CONSOLE_BRIDGE_logError("MoveitStateAdapter.isInCollision: Failed to secure a locked planning scene. Has the planing scene monitor been correctly initialized?");
```

Comment:
This function gets called repeatedly by Descartes for every waypoint during the graph building stage (which can take minutes even) and given this setup there's a chance that the PlanningScene instance held within the PSM changes as the graph is built resulting in nodes that are evaluated with different planning scenes.  While that mode of operation may be desirable for dynamic planning applications it does not apply for situations where one just wants to do offline planning on a static scene.  

I agree that the current collision capability is lacking in that it only accounts for the robot model in isolation and having the ability to include other environment objects( attached geometries, occupancy grids, .etc) is much needed but I think in the context of offline planning the scene should remain static.  

Could we maybe setup a way to add a custom collision callback that overrides a default collision function? The default could do collision checking against a static scene, while in a custom cb one could include a pointer to an active PSM that performs collision checks against an up-to-date environment.